### PR TITLE
Bug 1608103 - Fix dummy secrets generation

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Depends on https://phabricator.services.mozilla.com/D65084. 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: taskcluster/fenix_taskgraph/__init__.py [hunk 1]
```diff
@@ -36,6 +36,9 @@ def _import_modules(modules):
 
 
 def get_decision_parameters(graph_config, parameters):
+    parameters["tasks_for"] = "cron"
```
Comment (JohanLorenzo) [inline taskcluster/fenix_taskgraph/__init__.py]: TODO: These 2 lines are just for testing purpose and will be removed before landing the patch

File: taskcluster/fenix_taskgraph/transforms/build.py [hunk 2]
```diff
@@ -50,15 +51,16 @@ def add_shippable_secrets(config, tasks):
...
-            ] for fake_value, target_file in (
-                ("--", ".adjust_token"),
-                ("", ".digital_asset_links_token"),
-                ("-:-", ".leanplum_token"),
-                ("", ".mls_token"),
+            dummy_secrets.extend([{
+                "content": fake_value,
+                "path": target_file,
+            } for fake_value, target_file in (
+                ("faketoken", ".adjust_token"),
```
Comment (JohanLorenzo) [inline taskcluster/fenix_taskgraph/transforms/build.py]: I checked each real secret to look at what they really look like. The new ones are closer to real token than the former ones.

File: taskcluster/fenix_taskgraph/transforms/build.py [hunk 3]
```diff
@@ -50,15 +51,16 @@ def add_shippable_secrets(config, tasks):
                 ('mls', '.mls_token'),
             )])
         else:
-            task["run"]["pre-gradlew"] = [[
-                "echo", '"{}"'.format(fake_value), ">", target_file
```
Comment (JohanLorenzo) [inline taskcluster/fenix_taskgraph/transforms/build.py]: Bug: This command didn't work. I saw it when I backported the patch on reference-browser (https://github.com/mozilla-mobile/reference-browser/pull/1034 https://firefox-ci-tc.services.mozilla.com/tasks/YsO0TONlTF27dq5WKo38hw/runs/0/logs/https%3A%2F%2Ffirefox-ci-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FYsO0TONlTF27dq5WKo38hw%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log#L1407)

I looked closely into it and it turns out `shell_quote()`

https://github.com/mozilla-mobile/fenix/blob/f20fb504818726c7391c7b126e89d8a8ffea4381/taskcluster/fenix_taskgraph/job.py#L122

ensures `>` is quoted. This usually avoids security problems. Thus we do want to keep `shell_quote`. That's why I moved the fake token generation into its own script (`write-dummy-secret.py`). It's very similar to `get-secret.py`

Comment (MihaiTabara) [reply taskcluster/fenix_taskgraph/transforms/build.py]: Thanks for the context!

**Comment by JohanLorenzo** [comment]
Same comment as https://github.com/mozilla-mobile/fenix/pull/8046#issuecomment-593952059

**Pull request merged**